src/dispatch/plugins/dispatch_slack/modals/incident/views.py

```diff
@@ -0,0 +1,324 @@
+    """Builds all blocks required for updating the participant modal."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Edit Participant"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "mrkdwn",
+                        "text": "Use this form to edit why a particpant was added to this incident.",
+                    }
+                ],
+            },
+            participants_select_block(incident=incident, initial_option=participant),
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Submit"},
+        "callback_id": UpdateParticipantCallbackId.update_view,
+        "private_metadata": json.dumps(
+            {"incident_id": str(incident.id), "channel_id": str(incident.conversation.channel_id)}
+        ),
+    }
+
+    # we need to show the reason if we're updating
+    if participant:
+        modal_template["blocks"].append(
+            {
+                "block_id": UpdateParticipantBlockId.reason_added,
+                "type": "input",
+                "element": {
+                    "type": "plain_text_input",
+                    "multiline": True,
+                    "initial_value": participant.added_reason or "",
+                    "action_id": UpdateParticipantBlockId.reason_added,
+                },
+                "label": {"type": "plain_text", "text": "Reason Added"},
+            }
+        )
+
+        modal_template["callback_id"] = UpdateParticipantCallbackId.submit_form
+
+    return modal_template
+
+
+def update_notifications_group(incident: Incident, db_session: SessionLocal):
+    """Builds all blocks required to update the membership of the notifications group."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Update Group Membership"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "plain_text",
+                        "text": "Use this form to update the membership of the notifications group.",
+                    }
+                ],
+            },
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Update"},
+        "callback_id": UpdateNotificationsGroupCallbackId.submit_form,
+        "private_metadata": json.dumps(
+            {"incident_id": str(incident.id), "channel_id": incident.conversation.channel_id}
+        ),
+    }
+
+    group_plugin = plugin_service.get_active_instance(
+        db_session=db_session, project_id=incident.project.id, plugin_type="participant-group"
+    )
+    members = group_plugin.instance.list(incident.notifications_group.email)
```

Comment:
We should check if the plugin is enabled before listing the members

Reply:
Will handle in a seperate PR